app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1681,6 +1683,99 @@ impl GraphEditorModel {
         find_free_place(starting_from, direction, node_areas)
     }
 
+    pub fn new_node_position_after_dropping_edge(
+        &self,
+        edge_id: EdgeId,
+        mouse_position: Vector2,
+    ) -> Vector2 {
+        let edge_source_node_id = self.edge_source_node_id(edge_id);
+        let edge_source_node = edge_source_node_id.and_then(|id| self.nodes.get_cloned_ref(&id));
+        self.new_node_position_based_on_mouse(mouse_position, edge_source_node)
+    }
+
+    pub fn new_node_position_based_on_mouse_without_source(
+        &self,
+        mouse_position: Vector2,
+    ) -> Vector2 {
+        let nearest_node = self.find_nearest_node(mouse_position);
+        self.new_node_position_based_on_mouse(mouse_position, nearest_node)
+    }
+
+    fn find_nearest_node(&self, mouse_position: Vector2) -> Option<Node> {
+        let mut min_distance_squared = f32::MAX;
+        let mut nearest_node = None;
+        let nodes = self.nodes.all.raw.borrow();
+        for node in nodes.values() {
+            // For calculating the approximate distance between a node and the mouse pointer, one
+            // of several possible reference points on a node can be chosen. In order of increasing
+            // complexity and precision:
+            //  a) node.position(),
+            //  b) the geometrical center of the node's area,
+            //  c) a point closest to the mouse pointer on the line segment connecting node's
+            //     leftmost & rightmost points,
+            //  d) a point closest to the mouse pointer on the bounding box of the node
+            //     (with extra handling for when the mouse is inside the bounding box),
+            //  e) a point closest to the mouse pointer on the rounded rectangle forming the node's
+            //     shape (with extra handling for when the mouse is inside the rounded rectangle).
+            //  The following code implements variant (c), the first of the listed ones that
+            //  resulted in a UX which feels reasonably predictable and intuitive.
+            let node_position = node.position().xy();
```

Comment:
This documentation is poorly written and it not understandable. It needs to be rewritten from scratch. Please apply that to other docs too. Just a few notes about it:
1. `In order of increasing complexity [...]`: how did you test the complexity? Is it by the amount of code to be written / how hard the math formulas are there, or anything else? E.g. I do not see point b) as harder than point a). 

2. `In order of increasing [...] precision`: this is completely not understandable. Why for example solution a) is less precise then solution b)?

3. `on the line segment connecting node's leftmost & rightmost points` - what are these points? They are not explained.

4. Point d) describes `(with extra handling for when the mouse is inside the bounding box)` – why other points do not describe similar situations, like `(with extra handling when the point is on the line)` for point c)? Also, what does this information brings to the reader? The whole enumeration is about possibilities, and now this point describes implementation details.

5. `resulted in a UX which feels reasonably predictable and intuitive` – this does not tell anything to the reader. If you want to compare solutions, do it properly and be very precise of what is good / wrong with them.

6. I do not think solution c) is right (does not even matter what these points are) – have you tested it with expanded nodes with visualizations? Nodes can be higher than wider.

Reply:
I removed this comment block, added a detailed explanation to `find_nearest_node` documentation, and modified the code to take into account rounded corners radius.

I believe the solution is now right as long as we don't take into account visualizations. The design doc does currently seem to use the term *"the closest node [to the mouse pointer]"* without defining more precisely how to understand it, and whether a visualization is considered part of the node for purpose of this calculation. It does not define how to resolve the cases when nodes areas overlap either (i.e., which node is "the closest node to the mouse pointer" when 2+ nodes overlap and the mouse pointer is positioned in this overlapping area) - this is a corner case that can happen both when nodes are positioned using the GUI, as well as when raw metadata in an `*.enso` file is modified outside GUI. The current code handles this situation. If the current code is not right in your understanding, could you please describe an algorithm that should be used here, that would be right, and that would handle such corner cases?